bika/lims/databasesanitize/controller_view.py

```diff
@@ -0,0 +1,63 @@
+import json
+
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from collective.taskqueue.interfaces import ITaskQueue
+from plone.protect import CheckAuthenticator
+from zope.component import queryUtility
+
+from bika.lims import logger
+from bika.lims.browser import BrowserView
+from bika.lims.databasesanitize.analyses import analyses_creation_date_recover
+
+
+class ControllerView(BrowserView):
+    """
+    This class is the controller for the sanitize view. Sanitize view
+    """
+    template = ViewPageTemplateFile("templates/main_template.pt")
+
+    def __call__(self):
+        # Need to generate a PDF with the stickers?
+        if self.request.form.get('analysis_date', '0') == '1':
+            logger.info("Starting sanitation process 'Analyses creation "
+                        "date recover'...")
+            # Only load asynchronously if queue sanitation-tasks is available
+            task_queue = queryUtility(ITaskQueue, name='sanitation-tasks')
+            if task_queue is None:
+                # sanitation-tasks queue not registered. Proceed synchronously
+                logger.info("SYNC sanitation process started...")
+                task_manager = AsyncAnalysesCreationDateRecover()
```

Comment:
Since you are firing this task synchronously, using a class with name `AsyncAnalysesCreationDateRecover` is a bit confusing. In fact, this `AsyncAnalysesCreationDateRecover` object does nothing asynchronously, so I'd replace it's name by `AnalysesCreationDateRecover`. Remember to apply the changes to the .zcml too.